rails/config/database.yml

```diff
@@ -96,8 +95,10 @@ staging:
 #
 #   production:
 #     url: <%= ENV['DATABASE_URL'] %>
-#
+
 production:
   <<: *default
   url: <%= ENV.fetch('DATABASE_URL', '').sub(/^postgres/, "postgis") %>
   schema_search_path: public,heroku_ext
+  username: <%= ENV.fetch('POSTGRES_USER') { nil } %>
```

Comment:
username is already contained in the database url. I believe there is no need to set username here.